packages/app-extensions/src/rest/helpers.js

```diff
@@ -15,31 +17,50 @@ import {call} from 'redux-saga/effects'
  * @param transformer {function} Function to directly manipulate the result. By default returns data attribute
  */
 export function* fetchEntity(entityName, key, options, transformer = json => json) {
-  const queryParams = yield call(buildParams, options)
+  const queryParams = yield call(buildEntityQueryObject, options)
   const resp = yield call(requestSaga, `entities/${entityName}/${key}`, {queryParams})
   return yield call(transformer, resp.body)
 }
 
+/**
+ * Fetch amount of entities
+ *
+ * @param entityName {String} Name of the entity
+ * @param options {Object} see 'buildEntityQueryObject' function
+ * @param endpoint {String} To overwrite default endpoint entities/entityName/search
+ */
+export function* fetchEntityCount(entityName, options, endpoint, method = 'POST') {
+  const params = buildEntityQueryObject(options)
+  const resource = (endpoint || `entities/${entityName}`) + '/count'
+
+  const requestOptions = {
+    method,
+    ...(method === 'GET' ? {queryParams: queryObjectToUrlParams(params)} : {body: params})
+  }
+
+  const response = yield call(requestSaga, resource, requestOptions)
+  return response.body.count
+}
+
 /**
  * Helper to fetch entities.
  *
  * @param entityName {String} Name of the entity
- * @param options {Object} An object which can contain the following options:
- * - page {Number} Current page, needed to calculate offset. Default is undefined which means no offset.
- * - limit {Object} Maximum amount of entities to be retrieved
- * - orderBy {Array} Array of objects containing a field and order String. e.g. [{field: 'firstname', order: 'asc'}]
- * - paths {Array} Paths that should be returned
- * - fields {Array} Fields that should be returned
- * - relations {String} Relations that should be returned
- * - searchFilters {Array} List of filter names to be applied
- * - searchInputs {Object} Object that can contain search field values and fulltext _search attribute
- * - formName {String} Name of the form, needed to resolve display expressions and such
- * - query {String} TQL query
+ * @param options {Object} see 'buildEntityQueryObject' function
  * @param transformer {function} Function to directly manipulate the result. By default returns data attribute
+ * @param endpoint {String} To overwrite default endpoint entities/entityName/search
+ * @param method {String} To overwrite default http method 'POST'
  */
-export function* fetchEntities(entityName, options, transformer = defaultEntityTransformer) {
-  const queryParams = yield call(buildParams, options)
-  const resp = yield call(requestSaga, `entities/${entityName}`, {queryParams})
+export function* fetchEntities(entityName, options, transformer = defaultEntityTransformer, endpoint, method = 'POST') {
```

Comment:
- quite a lot method parameters. shouldn't become any more than 5, IMO
- or is there a way to reduce the number of parameters?
- can we rename `options` to something a little bit more specific?